project_stock/models/stock_move.py

```diff
@@ -0,0 +1,127 @@
+# Copyright 2022 Tecnativa - Víctor Martínez
+# License AGPL-3.0 or later (http://www.gnu.org/licenses/agpl)
+from odoo import _, api, exceptions, fields, models
+
+
+class StockMove(models.Model):
+    _inherit = "stock.move"
+
+    task_id = fields.Many2one(
+        comodel_name="project.task",
+        string="Task for finished products",
+        check_company=True,
+    )
+    raw_material_task_id = fields.Many2one(
+        comodel_name="project.task", string="Task for material", check_company=True
+    )
+    is_done = fields.Boolean(
+        string="Done",
+        compute="_compute_is_done",
+        store=True,
+        help="Technical Field to order moves",
+    )
+
+    @api.depends("state")
+    def _compute_is_done(self):
+        for move in self:
+            move.is_done = move.state in ("done", "cancel")
+
+    @api.model
+    def default_get(self, fields_list):
+        defaults = super().default_get(fields_list)
+        if self.env.context.get("default_raw_material_task_id"):
+            task = self.env["project.task"].browse(
+                self.env.context.get("default_raw_material_task_id")
+            )
+            defaults["group_id"] = task.procurement_group_id.id
+            defaults["reference"] = task.name
+        return defaults
+
+    def _decrease_reserved_quanity(self, quantity):
+        """ Decrease the reservation on move lines but keeps the
+        all other data.
+        """
+        move_line_to_unlink = self.env["stock.move.line"]
+        for move in self:
+            reserved_quantity = quantity
+            for move_line in move.move_line_ids:
+                if move_line.product_uom_qty > reserved_quantity:
+                    move_line.product_uom_qty = reserved_quantity
+                else:
+                    move_line.product_uom_qty = 0
+                    reserved_quantity -= move_line.product_uom_qty
+                if not move_line.product_uom_qty and not move_line.qty_done:
+                    move_line_to_unlink |= move_line
+        move_line_to_unlink.unlink()
+        return True
+
+    def _action_assign(self):
+        res = super()._action_assign()
+        for move in self.filtered(lambda x: x.task_id or x.raw_material_task_id):
+            if move.move_line_ids:
+                task = move.task_id if move.task_id else move.raw_material_task_id
+                move.move_line_ids.write({"task_id": task.id})
```

Comment:
Why do you need to pass this field to sml?

Reply:
I have removed this and defined the field as compute store.